pkg/xray/util.go

```diff
@@ -34,3 +36,14 @@ var getProjectKeySchema = func(isForceNew bool, additionalDescription string) ma
 		},
 	}
 }
+
+var resourceImporterForProjectKey = func(_ context.Context, d *schema.ResourceData, meta any) ([]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), ":", 2)
+
+	if len(parts) == 2 && parts[0] != "" && parts[1] != "" {
+		d.SetId(parts[0])
+		d.Set("project_key", parts[1])
+	}
+
+	return []*schema.ResourceData{d}, nil
```

Comment:
Just for my understanding:

If its not a project, nothing is changed to the state, but then it returns an array of schema.resourceData pointers? Is that just because `StateContext` expects an array?

Reply:
Yes, this func allows modification of the state data during import. So it expects the return to contain the state to be used. Why an array? Because import (in general) also support multiple resources import so the array accommodates that scenario.